toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapView.kt

```diff
@@ -223,6 +227,28 @@ private fun ViewpointUpdater(
     }
 }
 
+/**
+ * Sets up the ViewpointChangedState's property and event.
+ *
+ * @since 200.4.0
+ */
+@Composable
+private fun ViewpointChangedEventHandler(mapView: MapView, viewpointChangedEvents: Map<ViewpointType, ((Viewpoint) -> Unit)>?) {
+    LaunchedEffect(viewpointChangedEvents) {
+        launch {
+            mapView.viewpointChanged.collect {
+                viewpointChangedEvents?.forEach {
+                    it.value.invoke(
+                        mapView.getCurrentViewpoint(
+                            it.key
+                        ) ?: throw IllegalStateException("Viewpoint should not be null")
+                    )
+                }
+            }
+        }
+    }
+}
```

Comment:
You should do a null check on `viewpointChangedEvents` before launching a coroutine. If `viewpointChangedEvents` there is no point in launching a coroutine. Also, I don't think there is a need for launchin a separate coroutine here, since you only want to collect a single event in this LaunchedEffect:
```suggestion
private fun ViewpointChangedEventHandler(mapView: MapView, viewpointChangedEvents: Map<ViewpointType, ((Viewpoint) -> Unit)>?) {
   viewpointChangedEvents?.let { lambdas ->
      mapView.viewpointChanged.collect {
         lambdas.foreach {
            it.value.invoke(
                        mapView.getCurrentViewpoint(
                            it.key
                        ) ?: throw IllegalStateException("Viewpoint should not be null")
                    )
         }
      }
   }
}
```